solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -116,6 +117,27 @@ public CoreAdminHandler(final CoreContainer coreContainer) {
     this.coreAdminAsyncTracker = new CoreAdminAsyncTracker();
   }
 
+  private static final ThreadLocal<Integer> RECONCILE_THRESHOLD = new ThreadLocal<>();
+
+  private static final Closeable CLEAR_RECONCILE_THRESHOLD = RECONCILE_THRESHOLD::remove;
+
+  public static Closeable setReconcileThreshold(String spec) {
+    if (spec != null && !spec.isEmpty()) {
+      int size = Integer.parseInt(spec);
+      if (size >= 0) {
+        RECONCILE_THRESHOLD.set(size);
+      } else if (size == -1) {
+        // support special value `-1` to effectively just print log messages
+        RECONCILE_THRESHOLD.set(Integer.MAX_VALUE);
+      }
+    }
+    return CLEAR_RECONCILE_THRESHOLD;
+  }
+
+  public static Integer getReconcileThreshold() {
```

Comment:
Should this actually be used somewhere or am I missing something?

Reply:
It's used from a plugin (the SizeAware Directory impl). So yes, not in anything that's.a part of this PR!